config.go

```diff
@@ -554,6 +554,76 @@ func (s *Settings) GetGithubAppAuth() *githubAppAuth {
 	}
 }
 
+// HasAppInstalled returns true if the GitHub app is installed on the repo.
+func (s *Settings) HasAppInstalled(ctx context.Context, owner, repo string) bool {
+	if owner == "" || repo == "" {
+		grip.Debug(message.Fields{
+			"message": "owner/repo not specified",
+			"caller":  "HasAppInstalled",
+			"owner":   owner,
+			"repo":    repo,
+			"ticket":  "EVG-19966",
+		})
+		return false
+	}
+
+	authFields := s.GetGithubAppAuth()
+	if authFields == nil {
+		grip.Debug(message.Fields{
+			"message": "could not parse key from PEM",
+			"caller":  "HasAppInstalled",
+			"owner":   owner,
+			"repo":    repo,
+			"ticket":  "EVG-19966",
+		})
+		return false
+	}
+
+	httpClient := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(httpClient)
+
+	key, err := jwt.ParseRSAPrivateKeyFromPEM(authFields.privateKey)
+	if err != nil {
+		grip.Debug(message.WrapError(err, message.Fields{
+			"message": "could not parse key from PEM",
+			"caller":  "HasAppInstalled",
+			"owner":   owner,
+			"repo":    repo,
+			"appId":   authFields.AppId,
+			"ticket":  "EVG-19966",
+		}))
+		return false
+	}
+
+	itr := ghinstallation.NewAppsTransportFromPrivateKey(httpClient.Transport, authFields.AppId, key)
+	httpClient.Transport = itr
+	client := github.NewClient(httpClient)
+	installationId, _, err := client.Apps.FindRepositoryInstallation(ctx, owner, repo)
```

Comment:
Does this use our github API limit? Calling this every time we need a token is going to be really expensive. In general this will probably be time consuming also -- should we be caching the installationId or is this likely to change?